### DISCUSSION TITLE
[Question][Hw2]

### DISCUSSION THREAD

**Discussion by anisrashidov** [prosyslab-classroom/cs348-information-security#683] [Q&A]
Name: Anis Rashidov.

Just to confirm that I understood the material. My question is that snippet#1 below is AES for only one block encryption, while snippet#2 is AES for the whole message, right? Also, should we consider space (' ') as a character and encrypt it as well, or should we simply skip it?
Thanks in advance.

![image](https://user-images.githubusercontent.com/89933766/229271405-0acf29fd-d592-4169-8d1e-90e8b1aface0.png)
![image](https://user-images.githubusercontent.com/89933766/229271422-be1b5fe5-911e-44ad-8a63-950e8609cb0c.png)

**Comment by KAIST-JongchanPark** [comment]
1. Yes
2. You should encrypt any character.(i.e. includes space, newline, ...)

**Comment by bongjunj** [comment]
1. AES module enc/decrypts a block. ECB, CBC, and CTR defines how to enc/decrypt a list of block using AES.
2. Every character is encoded into a byte (or bytes). And there are many ways to encode a character. The popular one is ASCII and UTF-8. Note that an encryption scheme is character-encoding agnostic.

**Comment by anisrashidov** [comment]
> 1. AES module enc/decrypts a block. ECB, CBC, and CTR defines how to enc/decrypt a list of block using AES.
> 2. Every character is encoded into a byte (or bytes). And there are many ways to encode a character. The popular one is ASCII and UTF-8. Note that an encryption scheme is character-encoding agnostic.

Don't we use ASCII in this assignment?

**Comment by tanapthetimid** [comment]
It doesn't matter what encoding we use here because we are encrypting the raw byte data.

**Comment by anisrashidov** [comment]
> It doesn't matter what encoding we use here because we are encrypting the raw byte data.

Yeah, but what about converting human-readable text to binary string?

**Comment by KAIST-JongchanPark** [comment]
We usually encrypt a plaintext with human-readable characters.

However, in general, enc/dec mechanism doesn't care about whether input plaintext is human-readable or not.
